fdbserver/KeyValueStoreMemory.actor.cpp

```diff
@@ -462,19 +463,19 @@ class KeyValueStoreMemory final : public IKeyValueStore, NonCopyable {
 		return total;
 	}
 
-	// Data format for normal operation:
+	// Data format for normal operation (1 byte padding at the end):
 	// +-------------+-------------+-------------+--------+--------+
 	// | opType      | len1        | len2        | param2 | param2 |
 	// | sizeof(int) | sizeof(int) | sizeof(int) | len1   | len2   |
 	// +-------------+-------------+-------------+--------+--------+
 	//
-	// However, if the operation is encrypted:
-	// +-------------+-------------+-------------+---------------------------------+-------------+--------+--------+
-	// | OpEncrypted | len1        | len2        | BlobCipherEncryptHeader         | opType      | param1 | param2 |
-	// | sizeof(int) | sizeof(int) | sizeof(int) | sizeof(BlobCipherEncryptHeader) | sizeof(int) | len1   | len2   |
-	// +-------------+-------------+-------------+---------------------------------+-------------+--------+--------+
-	// |                                plaintext                                  |           encrypted           |
-	// +-----------------------------------------------------------------------------------------------------------+
+	// However, if the operation is encrypted (1 byte padding at the end):
+	// +-------------+-------------+-------------+---------------------------------------------+-------------+---
+	// | OpEncrypted | len1   | len2   | headerSize |BlobCipherEncryptHeader    | opType      | param1 | param2 |
```

Comment:
For better understanding: for encrypted mutation it seems we have two OpTypes stored in the header: `opEncrypted` and `opType` itself; can we get rid of one?

Reply:
The first op tells us if this entry is encrypted in which case special pre-processing is needed. The second is the actual op which we need to perform the operation. I'm not sure if there is a way to combine the two

Reply:
Yeah, my point was we are using 8 bytes for the opType, where even 4 bytes for opType seems like an overkill to me :)

Reply:
Yeah im not sure if there was an inital reason for making it 4 bytes as opposed to 2 or even 1

Reply:
not sure if we shall steal the highest byte from OpType, using one bit of it for isEncrypted , and the rest reserve for future flags. cc @etschannen

Reply:
Made the highest bit represent isEncrypted